core/src/get_similar_mail_provider.rs

```diff
@@ -0,0 +1,32 @@
+use levenshtein::levenshtein;
+
+use crate::syntax::SyntaxDetails;
+
+const MAIL_PROVIDERS: &[&str] = &["gmail", "yahoo"];
+
+pub fn get_similar_mail_provider(syntax: &mut SyntaxDetails) {
+	let domain: &str = syntax.domain.as_ref();
+	let mut domain_parts = domain.split('.');
+
+	let provider = domain_parts
+		.next()
+		.expect("We already checked that the syntax is valid");
+	let tld = domain_parts
+		.next()
+		.expect("We already checked that the syntax is valid");
+
+	for possible_provider in MAIL_PROVIDERS {
+		let distance = levenshtein(provider, possible_provider);
+
+		if distance < 3 {
+			// Return full address
+			syntax.suggestion = Some(format!(
```

Comment:
Did you manage to test this locally @Agreon?

I tried stuff like gmali, gmeil, etc, but it seems they are blacklisted by mailchecker

Reply:
I encountered the same problem :). An easy check is e.g. `@mail.com`